workitems/tests/workitems_tests/test_workitems.py

```diff
@@ -128,16 +130,22 @@ def test_input_pass(inputs, adapter):
 def test_input_fail(inputs, adapter):
     item = inputs.current
     assert item.state is None
+    assert item.exception is None
     assert len(adapter.releases) == 0
 
-    item.fail(ExceptionType.BUSINESS, "MY_ERR_CODE", "Something went oopsy")
+    exception = {
+        "type": ExceptionType.BUSINESS,
+        "code": "MY_ERR_CODE",
+        "message": "Something went oopsy",
+    }
+
+    item.fail(*exception.values())
```

Comment:
Careful with this one, as there's no guarantee that the `.values()` are returned in order compared to how the `exception` dictionary was declared. I think [`OrderedDict`](https://docs.python.org/3/library/collections.html#collections.OrderedDict) may save you here.